File: src/easy/#1047. Remove All Adjacent Duplicates In String.js
// https://leetcode.com/problems/remove-all-adjacent-duplicates-in-string/description/

function removeDuplicates(str) {
  const stack = [];

  for (let i = 0; i < str.length; i++) {
    // если равен, то удаляем последний элемент стека и идем на следующей элемент строки
    if (str[i] === stack[stack.length - 1]) {
      stack.pop();
    } else {
      stack.push(str[i]); // если последний элемент стека не равен текущего элемента строки, то закидываем в стек
    }
  }

  return stack.join("");
}

File: src/easy/#1047. Remove All Adjacent Duplicates In String.py
def removeDuplicates(s):
    stack = []

    i = 0

    while i < len(s):

        if len(stack) and stack[-1] == s[i]:
            stack.pop(-1)
        else:
            stack.append(s[i])

        i = i + 1
        print(i, stack)

    return "".join(stack)
